Add starter code

connorjayr opened the pull request:
Adds starter code for the web server, using `actix-web` as the server framework. Also adds SPA routing to the frontend and integrates ESLint and Prettier for code style enforcement.

connorjayr committed:
Add starter code

Adds starter code for the web server, using `actix-web` as the server framework. Also adds SPA routing to the frontend and integrates ESLint and Prettier for code style enforcement.

jputlock commented on server/src/main.rs:
```diff
@@ -1,3 +1,22 @@
+    HttpServer::new(|| {
+        App::new()
+            .wrap(Logger::default())
+            .service(routes::login::handler)
```
This is fine for now -- we may want to move responsibility of configuring groups of routes (such as a group containing `login` and `logout`) elsewhere using `.configure()`

jputlock replied:
Also if we want to move them into an `auth` scope, we can use `web::scope()`

connorjayr replied:
Agree with both of the above. I think routes with shared prefixes can even create a `Scope` within the module itself, and then the caller does not need to worry about any of the details.

jputlock commented on README.md:
```diff
@@ -1,46 +1,3 @@
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
-
-To learn React, check out the [React documentation](https://reactjs.org/).
+Sust is a modern, role-based murder mystery game built using Rust and React.
```
Please add a note that `cargo-watch` is required.

jputlock replied:
Also a note about the correct `node` version.

connorjayr replied:
Added both in https://github.com/connorjayr/sust/pull/1/commits/d8a12d4d9a91afd8b9d04e5c72fb81b382981dfc.

connorjayr committed:
Add README

Adds a basic README that contains instructions for local development and deployment.

jputlock reviewed (approved):
LGTM 👍
